FixMath.h

```diff
@@ -0,0 +1,1287 @@
+/*
+ * FixMath.h
+ *
+ * Copyright 2023, Thomas Combriat and the Mozzi team
+ *
+ * This file is part of Mozzi.
+ *
+ * Mozzi is licensed under a Creative Commons
+ * Attribution-NonCommercial-ShareAlike 4.0 International License.
+ *
+ */
+
+
+/** This file implements two fixed point number classes. These numbers can have a fractional
+    part but are actually standard integers under the hood which makes calculations with them
+    efficient on platforms which do not have a FPU like most micro-controllers. These numbers can be
+    signed (SFixMath) or unsigned (UFixMath).
+
+    A fixed point number has its range defined by the number of bits encoding the integer part (NI 
+    in the following) and its precision by the number of bits encoding the fractional part (NF). For UFixMath types, the integral part can hold values in [0,2^NI-1], for SFixMath types, the integral part can hold values in [-2^NI,2^NI-1]. The number of bits encoding the fractional can be considered as the precision of the number: given NF, the number of possible values in the  [0,1[ range will 2^NF. Hence, given NF, the resolution will be 1/(2^NF).
+
+    Under the hood, these types will keep track of the maximum possible value they might hold (this is the RANGE template parameter), and, if only *SAFE* operations (see below) are used, will automatically adjust there NI and NF to accomodate the result of a operation. It will also try not to promote there internal type when possible, assuming that you use the complete range of a given type.
+
+    The operations possible with these types can be divided into two categories:
+    - the operations between FixMath types are all safe (aka won't overflow) and are the only one included by default
+    - the operations between a FixMath and a native C type (int, float) are NOT safe and are not included by default. In order to activate them, you need to `#define FIXMATHUNSAFE` before including FixMath.h.
```

Comment:
To keep doxygen happy, I believe it will also be necessary to add
```
#if FOR_DOYGEN_ONLY
#define MOZZI_FIXMATH_UNSAFE
#endif
```
at the top of this file.

Reply:
done.